### ISSUE TITLE
Replace all Async calls by Bluebird promises

### ISSUE THREAD

**Issue by kimar** [Geofancy/cloud-app#23]
Phew. Please. Finally. Do. This.

**Comment by francisdb** [comment]
I like :-) But might have been better to switch to native Node.js 4.x promisses

**Comment by kimar** [comment]
Sure thing :+1: whatever comes first. Already using bluebird here, but would go to Node 4 promises too

**Comment by kimar** [comment]
@francisdb not sure if native promises also support this, but using `Bluebird

**Comment by francisdb** [comment]
Mongoose can use native promises. 

Mongoose.promise = Promise;

But q is fine.
Als check const, arrow functions and other es6 features

**Comment by kimar** [comment]
Yeah I'm definitely in for using ES6 features :+1:
Bluebird provides the advantage that it can be used together with any library using callbacks. (If they're not totally screwing up with them) Bit I'm open to using native promises if they're feasible to use and we're not using multiple promise libs at once... ¯\_(ツ)_/¯

**Comment by francisdb** [comment]
I expect all libraries to start supporting native promises but we're not there yet :-)

**Comment by kimar** [comment]
Yeah I'd ❤️ to see this happen, but until then there's probably always compromises to make :-(

**Comment by kimar** [comment]
Started work on this here: https://github.com/LocativeHQ/cloud-app/tree/feature/no-async

As it's quite a tough task I'm not expecting this to go all at once, but it would be cool to see this eventually succeed at some point. The code all over locative, in all of the apps is super ugly and legacy :-(
